app/src/main/java/io/homeassistant/android/launch/LaunchActivity.kt

```diff
@@ -0,0 +1,22 @@
+package io.homeassistant.android.io.homeassistant.android.launch
+
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import io.homeassistant.android.api.Session
+import io.homeassistant.android.onboarding.OnboardingActivity
+import io.homeassistant.android.webview.WebViewActivity
+
+
+class LaunchActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        if (Session.getInstance().url.isNullOrBlank()) {
+            startActivity(OnboardingActivity.newInstance(this))
+        } else {
+            startActivity(WebViewActivity.newInstance(this))
```

Comment:
For future, maybe a good idea to fire off token valid check here? That way we don't have to wait until the frontend requests a token but can make sure it's valid.

Reply:
Yes agreed. Could we enable [Github Project](https://github.com/home-assistant/home-assistant-android/projects) to list all the task ?
like listen [connection-status](https://developers.home-assistant.io/docs/en/frontend_external_bus.html#connection-status-update-connection-status) and display authentication screen again in case of `"auth-invalid" | "disconnected"`